core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java

```diff
@@ -413,4 +447,56 @@ public InstanceId getInstanceId() {
     return context.getInstanceID();
   }
 
+  @Override
+  public ServerId getServer(ServerTypeName type, String host, int port) {
+    final HostAndPort hp = HostAndPort.fromParts(host, port);
+    switch (type) {
+      case COMPACTOR:
+        return context.getServerIdResolver().resolveCompactor(hp.toString());
+      case MANAGER:
+        final ManagerServerId msi = context.getServerIdResolver().getManager();
+        if (msi.getHost().equals(host) && msi.getPort() == port) {
+          return msi;
+        } else {
+          return null;
+        }
+      case SCAN_SERVER:
+        return context.getServerIdResolver().resolveScanServer(hp.toString());
+      case TABLET_SERVER:
+        return context.getServerIdResolver().resolveTabletServer(hp.toString());
+      default:
+        throw new IllegalArgumentException("Unhandled server type: " + type);
+    }
+  }
+
+  @Override
+  public Set<ServerId> getServers(ServerTypeName type) {
+    return getServers(type, null);
+  }
+
+  @Override
+  public Set<ServerId> getServers(ServerTypeName type, Predicate<ServerId> test) {
+    final Set<ServerId> results = new HashSet<>();
+    switch (type) {
+      case COMPACTOR:
+        results.addAll(context.getServerIdResolver().getCompactors());
```

Comment:
Not sure this is worthwhile, wondering if would be useful to push the predicate down.  Please ignore this comment if it does not make sense.

```suggestion
        results.addAll(context.getServerIdResolver(test).getCompactors());
```